source/Engine.cpp

```diff
@@ -131,11 +151,36 @@ namespace {
 		}
 	}
 
+	template <class Type>
+	void Prune(list<Type> &objects)
+	{
+		for(auto it = objects.begin(); it != objects.end(); )
+		{
+			if(it->ShouldBeRemoved())
+				it = objects.erase(it);
+			else
+				++it;
+		}
+	}
```

Comment:
std::erase_if ?
https://en.cppreference.com/w/cpp/container/list/erase2

Reply:
That's the plan, but 
![image](https://github.com/user-attachments/assets/748fa8d7-50e1-43d8-bbdc-55758d0ef400)

Reply:
oh ouch

Reply:
remove_if ?

Reply:
I'd rather just do the refactor once. C++20 is coming soon anyway.